edit-post/index.js

```diff
@@ -23,19 +23,17 @@ export * from './api';
 
 // Configure moment globally
 moment.locale( dateSettings.l10n.locale );
-if ( dateSettings.timezone.string ) {
-	moment.tz.setDefault( dateSettings.timezone.string );
-} else {
-	const momentTimezone = {
-		name: 'WP',
-		abbrs: [ 'WP' ],
-		untils: [ null ],
-		offsets: [ -dateSettings.timezone.offset * 60 ],
-	};
-	const unpackedTimezone = moment.tz.pack( momentTimezone );
-	moment.tz.add( unpackedTimezone );
-	moment.tz.setDefault( 'WP' );
-}
+
+// create WP timezone based off dateSettings
+const offsets = ( dateSettings.timezone.string ) ?  moment.tz.zone( dateSettings.timezone.string ).offsets : [ -dateSettings.timezone.offset * 60 ];
+const momentTimezone = {
+	name: 'WP',
+	abbrs: [ 'WP' ],
+	untils: [ null ],
+	offsets: [ offsets ],
+};
+const unpackedTimezone = moment.tz.pack( momentTimezone );
+moment.tz.add( unpackedTimezone );
```

Comment:
Do you think we should move this to `date/index.js` instead, since we're already using moment?